src/SimpleAmqpClient/Channel.h

```diff
@@ -65,30 +65,36 @@ class SIMPLEAMQPCLIENT_EXPORT Channel : boost::noncopyable {
   static const std::string EXCHANGE_TYPE_TOPIC;
 
   /**
-    * Creates a new channel object
-    * Creates a new connection to an AMQP broker using the supplied parameters
+   * Creates a new channel object
+   * Creates a new connection to an AMQP broker using the supplied parameters
    * and opens
-    * a single channel for use
-    * @param host The hostname or IP address of the AMQP broker
-    * @param port The port to connect to the AMQP broker on
-    * @param username The username used to authenticate with the AMQP broker
-    * @param password The password corresponding to the username used to
+   * a single channel for use
+   * @param host The hostname or IP address of the AMQP broker
+   * @param port The port to connect to the AMQP broker on
+   * @param username The username used to authenticate with the AMQP broker
+   * @param password The password corresponding to the username used to
    * authenticate with the AMQP broker
-    * @param vhost The virtual host on the AMQP we should connect to
-    * @param channel_max Request that the server limit the number of channels
+   * @param vhost The virtual host on the AMQP we should connect to
+   * @param channel_max Request that the server limit the number of channels
    * for
-    * this connection to the specified parameter, a value of zero will use the
+   * this connection to the specified parameter, a value of zero will use the
    * broker-supplied value
-    * @param frame_max Request that the server limit the maximum size of any
+   * @param frame_max Request that the server limit the maximum size of any
    * frame to this value
-    * @return a new Channel object pointer
-    */
+   * @param heartbeat The number of seconds between heartbeat frames to
+   * request of the broker. A value of 0 disables heartbeats.
+   * Note that dependency, rabbitmq-c, only has partial support for heartbeats, as of
+   * v0.4.0.
+   * @return a new Channel object pointer
+   */
   static ptr_t Create(const std::string &host = "127.0.0.1", int port = 5672,
                       const std::string &username = "guest",
                       const std::string &password = "guest",
-                      const std::string &vhost = "/", int frame_max = 131072) {
+                      const std::string &vhost = "/",
+                      int frame_max = 131072,
+                      int heartbeat = 0) {
```

Comment:
Instead of adding another default parameter, create a new overload for Create with a heartbeat parameter, so the same for other public methods that heartbeat is being added to.